src/Accounting/index.ts

```diff
@@ -266,13 +266,12 @@ export const calcTradeExposureFromQuoteAndLeverage: (
  */
 export const calcSlippage: (
     orderAmount: BigNumber,
-    leverage: BigNumber,
     orders: FlatOrder[],
-) => { slippage: BigNumber, tradePrice: BigNumber} = (orderAmount, leverage, orders) => {
+) => { slippage: BigNumber, tradePrice: BigNumber} = (orderAmount, orders) => {
     if (orders.length) {
         // weighted average of the price, where the weights are the amounts at each price
         let
-            totalAmount = orderAmount.times(leverage),
+            totalAmount = Number.isNaN(orderAmount.toNumber()) ? new BigNumber(0) : orderAmount,
```

Comment:
this looks like it would work, but it looks indicative of a problem somewhere else. why would orderAmount ever not be a number?

Reply:
When you have an input field and nothing in it it sets the value to NaN. So if I had
<input value={exposure} onChange={(e) => setExposure(e.target.value)} /> or whatever, when it sets exposure it might go exposure = new BigNumber(valueToSet). If valueToSet is NaN the BigNumber doesnt throw any errors. But when you try to evaluate or use it it results in some funky stuff. So its if someone passes a bad parameter. Alternatively you could as you said you could do a conditional before the input comes in.

Reply:
But this is more robust in my opinion since the equation handles the inputs better

Reply:
it makes sense and happy to bang it in, but I would argue it's probably a better design to have a NaN handler in the frontend that is applied to all onChangeEvents on input, then you solve it everywhere in the frontend that numbers from inputs are used